### ISSUE TITLE
There is an empty buffer when `--remote` first starts Vim

### ISSUE THREAD

**Issue by roccomao** [vim/vim#13851]
**TLDR;** The root cause may be caused by the `:drop` command.

When we open Vim without a filename parameter, there is an unnamed empty `[No Name]` buffer. This is normal. When we pass the filename parameters, this empty buffer no longer exists. This is the normal working mode we expect.

But when `--remote` related parameters are included, there will always be a `[No Name]` buffer. To reproduce: Command line launch `gvim --remote-silent test.txt` when NO Vim exists; Or when starting Vim for the first time with a server name that does not currently exist like `gvim --servername NotExist --remote-silent test.txt`. Anyway, make sure you **start Vim** with the `--remote` parameter **for the first time**.

Know from the documentation:

>The `--remote` argument will cause a `:drop` command to be constructed from the rest of the command line and sent as described above.

So I started Vim directly, just entered `gvim` without any parameters, and then called the `:drop` command. The original `[No Name]` buffer will also not be reused.

--------

**Solutions I considered**

This may be solved directly in the code related to *clientserver*, but it may be better to directly modify the judgment logic of the `:drop` command. For the `:drop file` command, if the `file` has not been opened, at the same time, the current buffer is a `[No Name]` buffer and has not been modified yet, then reuse the current buffer just like `:edit file`. Maybe you have a better solution. Looking forward to your review and implementation, much appreciated.

--------

**Other Informations**

- OS: Microsoft Windows 10.0.19045
- Version of Vim: latest gVim 64-bit [v9.1.0026](https://github.com/vim/vim-win32-installer/releases/tag/v9.1.0026) from [vim-win32-installer](https://github.com/vim/vim-win32-installer)
